File: 3024.rb
# @param {Integer[]} nums
# @return {String}
def triangle_type(nums)
  return 'none' if nums.any? { |e| nums.sum - 2*e <= 0 }
  return 'equilateral' if nums.uniq.count == 1
  return 'isosceles' if nums.uniq.count == 2
  "scalene"
end

File: 3024.cpp
class Solution {
public:
  string triangleType(vector<int>& nums) {
    if (
      !(
        nums[0] < nums[1] + nums[2] &&
        nums[1] < nums[2] + nums[0] &&
        nums[2] < nums[0] + nums[1]
      )
    ) return "none";

    if (nums[0] == nums[1] && nums[1] == nums[2]) {
      return "equilateral";
    }

    if (
      nums[0] == nums[1] ||
      nums[1] == nums[2] ||
      nums[2] == nums[0]
    ) return "isosceles";
    return "scalene";
  }
};
